wordpress/wp-content/plugins/cds-base/classes/Modules/Site/SiteSettings.php

```diff
@@ -0,0 +1,124 @@
+
+            <form method="post" action="options.php" id="collection_settings_form" class="gc-form-wrapper">
+                <?php
+                settings_fields('site_settings_group');
+                do_settings_sections('collection-settings-admin');
+                submit_button();
+                ?>
+            </form>
+        </div>
+    <?php }
+
+    public function collectionSettingsPageInit()
+    {
+        register_setting(
+            'site_settings_group', // option_group
+            'collection_mode',
+            function ($input) {
+                if ($input !== "maintenance" || $input !== "live") {
+                    return "maintenance";
+                }
+
+                return sanitize_text_field($input);
+            }
+        );
+
+        register_setting(
+            'site_settings_group', // option_group
+            'collection_mode_maintenance_page',
+        );
+
+        add_settings_section(
+            'collection_settings_section', // id
+            get_bloginfo("name"), // title
+            null, // callback
+            'collection-settings-admin' // page
+        );
+
+        add_settings_field(
+            'collection_mode', // id
+            _('Mode', 'cds-snc'), // title
+            array( $this, 'collectionModeCallback'), // callback
+            'collection-settings-admin', // page
+            'collection_settings_section', // section
+            [
+                'label_for' => 'collection_mode'
+            ]
+        );
+
+        add_settings_field(
+            'collection_mode_maintenance_page', // id
+            _('Maintenance Page', 'cds-snc'), // title
+            array( $this, 'collectionMaintenancePageCallback'), // callback
+            'collection-settings-admin', // page
+            'collection_settings_section', // section
+            [
+                'label_for' => 'collection_mode_maintenance_page'
+            ]
+        );
+    }
+
+    public function collectionModeCallback()
+    {
+        $collection_mode = get_option('collection_mode');
+
+        printf('<input type="radio" name="collection_mode" id="collection_maintenance" value="maintenance" %s /> <label for="collection_maintenance">Maintenance</label><br />', checked('maintenance', $collection_mode, false));
+        printf('<input type="radio" name="collection_mode" id="collection_live" value="live" %s /> <label for="collection_live">Live</label><br />', checked('live', $collection_mode, false));
+    }
+
+    public function collectionMaintenancePageCallback()
+    {
+        wp_dropdown_pages(
+            array(
+                'name'              => 'collection_mode_maintenance_page',
+                'echo'              => 1,
+                'show_option_none'  => __('&mdash; Select &mdash;'),
+                'option_none_value' => '0',
+                'selected'          => get_option('collection_mode_maintenance_page'),
+            )
+        );
+    }
+}
```

Comment:
This is way smarter than trying to do something really customizable. (ie, add a bunch of weird options to the settings page ("Page title", "Page content", etc).